server/server_dynamic_proxy.go

```diff
@@ -36,31 +36,50 @@ func (s *Server) getCookieHandler(w http.ResponseWriter, r *http.Request) []byte
 	return []byte(cookie.Value)
 }
 
-// handleDynamicProxy is the main http websocket handler for the chisel server
-func (s *Server) handleDynamicProxy(w http.ResponseWriter, r *http.Request) bool {
-	res, _ := httputil.DumpRequest(r, true)
+func (s *Server) authorizeRequest(w http.ResponseWriter, r *http.Request, op, proxyId string) []byte {
 
 	ua := r.Header.Get("User-Agent")
 	host := r.Host
-	s.Infof(string(res))
+	// s.Infof(string(res))
 	// Do a client authentication.
 	authKey := []byte(r.Header.Get("Authorization"))
 	if len(authKey) == 0 {
 		authKey = s.getCookieHandler(w, r)
 		if len(authKey) == 0 {
-			return true
+			s.Infof("Authorization key not found.")
+			http.Error(w, s.Errorf("Authorization key not found.").Error(), http.StatusUnauthorized)
+			return nil
+		}
+	}
+	// s.Infof("Authorizing user key: %v, host: %v, ua: %v", string(authKey), host, ua)
+	if op != "/register" && op != "/unregister" {
+		// check cache then authorize and return key
+		cachedAuthKey := []byte{}
```

Comment:
i agree, i have these changes in progress. the reason i didnt do this before was because register would be called from api, which would already do this check. i think we can skip this for now.

Reply:
wait there are 2 parts to it. one when registering we dont need to check assuming api already does the check. but when executing we need to check if a job to the same cluster ip port exists. but on the otherhand again adding this logic to a reverseproxy seems too specific,  I would rather add deletion of the proxy if the job dies or is stopped.